dace/codegen/cppunparse.py

```diff
@@ -320,32 +320,35 @@ def _Assign(self, t):
                 (ast.Subscript, ast.Attribute)) and not self.locals.is_defined(
                     target.id, self._indent):
 
-                # the target is not already defined: we should try to infer the type
-                if self.type_inference is True:
-                    # Perform type inference
-                    # Build dictionary with symbols
-                    def_symbols = {}
-                    def_symbols.update(self.locals.get_name_type_associations())
-                    def_symbols.update(self.defined_symbols)
-                    inferred_symbols = type_inference.infer_types(
-                        t, def_symbols)
-                    inferred_type = inferred_symbols[target.id]
-
-                    self.locals.define(target.id, t.lineno, self._indent,
-                                       inferred_type)
-                    if self.language == dace.dtypes.Language.OpenCL and (
-                            inferred_type is not None
-                            and inferred_type.veclen > 1):
-                        # if the veclen is greater than one, this should be defined with a vector data type
-                        self.write("{}{} ".format(
-                            dace.dtypes._OCL_VECTOR_TYPES[inferred_type.type],
-                            inferred_type.veclen))
+                # if the target is already defined, do not redefine it
+                if self.defined_symbols is None or target.id not in self.defined_symbols:
+                    # we should try to infer the type
+                    if self.type_inference is True:
+                        # Perform type inference
+                        # Build dictionary with symbols
+                        def_symbols = {}
+                        def_symbols.update(
+                            self.locals.get_name_type_associations())
+                        def_symbols.update(self.defined_symbols)
+                        inferred_symbols = type_inference.infer_types(
+                            t, def_symbols)
+                        inferred_type = inferred_symbols[target.id]
+
+                        self.locals.define(target.id, t.lineno, self._indent,
+                                           inferred_type)
+                        if self.language == dace.dtypes.Language.OpenCL and (
+                                inferred_type is not None
+                                and inferred_type.veclen > 1):
+                            # if the veclen is greater than one, this should be defined with a vector data type
+                            self.write("{}{} ".format(
+                                dace.dtypes._OCL_VECTOR_TYPES[
+                                    inferred_type.type], inferred_type.veclen))
+                        else:
+                            self.write(dace.dtypes._CTYPES[inferred_type.type] +
+                                       " ")
```

Comment:
How is this related? Maybe add a comment about it on the PR.

Reply:
Yes. It popped while I was changing the test (but then i changed it again). Will add a comment